composer.json

```diff
@@ -47,6 +48,8 @@
     },
     "prefer-stable": true,
     "scripts": {
+        "compile": "classpreloader.php compile --config=./bridge/_files.php --output=./bridge/_generated.php --strip_comments=1",
```

Comment:
I think one of these [`composer` events](https://getcomposer.org/doc/articles/scripts.md#event-names) can be hooked to call the compile script. We need to remove `autoload: {"psr-4": ...}`, as we aren't only a PSR-4 autoloader (because we have functions to load).

Reply:
Retracting my 👍 here as well. Or at least I want to expose my reasoning first.

> I think one of these composer events can be hooked to call the compile script.

I assume that you would like to have our users to use the generate script instead of the standard autoloading mechanism used by composer. In this case I have a soft disagreement.

1. If we have our users to generate the script and something goes wrong with the script generation be blocks development. We are starting only now using this library, theoretically we should have reproducible builds but I'd feel more comfortable in having users that requires manual instrumentation to still use the composer autoloader, while users not requiring manual instrumentation (hence using our code) to use our provided generated file that is TESTED in CI. This at the cost of some performance ONLY for the classes that they need and we do not use.
2. If in a few months we observe zero issues in CI with this new approach, I will be more than open to make generate the compiled file as part of a post-X hook.

> We need to remove autoload: {"psr-4": ...}, as we aren't only a PSR-4 autoloader (because we have functions to load).

We have not functions to load as part of this PR. This PR, from the user standpoint, it is still a fully compatibile psr4 autoload. Unless I am missing something.

Reply:
1. The only way I can think of right now to avoid the situation where the user's script doesn't compile the generated file is to ship it in git, as composer uses the release tag and not the uploaded builds.
2. Fair.
3. This is true, but I thought we were going to use the compile-to-single-file PR to fix the remaining testing issues that occur in optimize/config. In that PR we are no longer a PSR-4 compatible autoloader and using a single file as our autoloader should fix the issues.